### PR TITLE
Xen pvh

### PR DESCRIPTION
**Pull request by hannesm**
the changes are picked from @talex5 master branch and @yomimono repository

### CODE REVIEW

**Comment by hannesm** [comment]
I suggest to keep `master` in sync with upstream (xen), and do releases to opam repository from `active` (of course, if there's a better name suggestion, please comment).

**Comment by hannesm** [comment]
the patches from talex5/master not included in this PR are:
- 818ae557446fd7a817714d67b9c44cf544e1100e (printf for float)
- 744a56203c585c979dd2175afba9407fabf9cbaf (console output, block and retry)

plus the arm-specific changes (which as far as I know is not used by anyone, and thus I didn't bother rebasing them).

File: console/console.c [hunk 1]
```diff
@@ -174,4 +174,13 @@ void resume_console(void)
...
\ No newline at end of file
+}
+
+__attribute__((weak)) void minios_show_banner(void)
+{
+#ifdef CONFIG_PARAVIRT
+    printk("Xen Minimal OS (pv)!\n");
+#else
+    printk("Xen Minimal OS (hvm)!\n");
+#endif
```
Comment (cfcs) [inline console/console.c:185]: Where does PVH fit in this?

Comment (hannesm) [reply console/console.c:185]: this change is part of https://github.com/mirage/mini-os/pull/6/commits/b0ec3d00a88880c85b9a9499c9dcf11365fec4fe -- the string is as-is from upstream. We can change this (in a separate commit, maybe later?), but I'd keep this separate from the commit above. If you feel strongly about this, why not submit a PR to this PR?

**Comment by hannesm** [comment]
superseeded by https://github.com/mirage/mirage/issues/1159

**Pull request closed without merging**